### PR TITLE
CAMEL-17173 Mute server errors by default

### PR DESCRIPTION
**Pull request by jono-morris**
Updated code so that by default we don't include stack traces in the response body when we returning HTTP 500.  

Also added a new option `logException` to log exceptions when a muted response, one not containing a stack trace, is returned.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java [hunk 1]
```diff
@@ -41,13 +41,29 @@ public void testMuteException() throws Exception {
...
 
+    @Test
+    public void testDefaultMuteException() throws Exception {
+        HttpGet get = new HttpGet("http://localhost:" + getPort() + "/fooDefault");
+        get.addHeader("Accept", "application/text");
+        try (CloseableHttpClient client = HttpClients.createDefault();
+             CloseableHttpResponse response = client.execute(get)) {
+
+            String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
+            assertEquals("", responseString);
```
Comment (orpiske) [inline components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java:52]: Maybe use `assertTrue(responseString.isEmpty())` ?

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java [hunk 2]
```diff
@@ -41,13 +41,29 @@ public void testMuteException() throws Exception {
...
     }
 
+    @Test
+    public void testDefaultMuteException() throws Exception {
+        HttpGet get = new HttpGet("http://localhost:" + getPort() + "/fooDefault");
+        get.addHeader("Accept", "application/text");
+        try (CloseableHttpClient client = HttpClients.createDefault();
+             CloseableHttpResponse response = client.execute(get)) {
+
+            String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
```
Comment (orpiske) [inline components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java:51]: Use `StandardCharsets.UTF_8` 

Obs.: I know the other test in there uses "UTF-8" (this is an old practice - we still have a few places doing this).

File: components/camel-servlet/src/test/java/org/apache/camel/component/servlet/ServletMuteExceptionTest.java [hunk 3]
```diff
@@ -25,6 +25,18 @@
...
+    @Test
+    public void testMuteDefaultTrue() throws Exception {
+        WebRequest req = new PostMethodWebRequest(
+                contextUrl + "/services/muteDefault",
+                new ByteArrayInputStream("".getBytes()), "text/plain");
+        WebResponse response = query(req, false);
+
+        assertEquals(500, response.getResponseCode());
+        assertEquals("text/plain", response.getContentType());
+        assertEquals("", response.getText());
```
Comment (orpiske) [inline components/camel-servlet/src/test/java/org/apache/camel/component/servlet/ServletMuteExceptionTest.java:37]: Same note I raised on another text.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for your contribution. I added a few requests.

**Comment by jono-morris** [comment]
Also noticed that I need to fix up a couple of the unit tests in camel-jetty that expect a response body with HTTP 500.

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java (hunk 1 above)
Comment (jono-morris) [reply components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java]: Done, thanks for that.

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java (hunk 2 above)
Comment (jono-morris) [reply components/camel-jetty/src/test/java/org/apache/camel/component/jetty/JettyMuteExceptionTest.java]: Done.

**Comment by davsclaus** [comment]
Such a change needs to be documented in the 4.1 upgrade guide in the docs folder

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
@jono-morris is all okay, did you fix those tests ?

**Comment by jono-morris** [comment]
> @jono-morris is all okay, did you fix those tests ?

@davsclaus, yes I've fixed all the tests.

**Pull request merged**